src/realm/sync/subscriptions.cpp

```diff
@@ -273,11 +275,24 @@ void MutableSubscriptionSet::check_is_mutable() const
     }
 }
 
+// This uses the 'swap and pop' idiom to run in constant time.
+// The iterator returned is:
+//  1. end(), if the last subscription is removed
+//  2. same iterator it is passed (but pointing to the last subscription in set), otherwise
 MutableSubscriptionSet::iterator MutableSubscriptionSet::erase(const_iterator it)
 {
     check_is_mutable();
     REALM_ASSERT(it != end());
-    return m_subs.erase(it);
+    if (it == std::prev(m_subs.end())) {
+        m_subs.pop_back();
+        return end();
+    }
+    auto back = std::prev(m_subs.end());
+    // const_iterator to iterator in constant time (See https://stackoverflow.com/a/10669041)
+    auto iterator = m_subs.erase(it, it);
+    std::swap(*iterator, *back);
```

Comment:
Doesn't this change the order of the subscription list? Tho I don't think we really care about subscription ordering?

Reply:
It does. We indeed don't care. I believe there is some sorting involved when sending it to the server.